post/views.py

```diff
@@ -21,22 +16,40 @@ class AddPost(CreateView):
 
     def form_valid(self,form):
         form.instance.owner = self.request.user
-        breakpoint()
         return super().form_valid(form)
-
-
+        
+class LikeList(ListView):
+    model = Like
+    template_name = "post_detail.html"
+    permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
+    
 class PostList(ListView):
     model = Post 
     template_name = 'post_list.html'
-    permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
 
+    permission_classes = (permissions.IsAuthenticatedOrReadOnly,) 
+
+    def get_context_data(self,**kwargs):
+        context = super().get_context_data(**kwargs)
+        l=[]
+        m = list(Like.objects.filter(user=self.request.user).values('post')) 
+
+        for i in m:
+            l.append(i.get('post'))
```

Comment:
[comment]
As an exercise, try rewriting this to use a list comprehension to compose `context['liked_posts']` in a single line!